students/FlorentinPopescu/07_Mailroom_Part1_L3_FP.py

```diff
@@ -0,0 +1,162 @@
+# -*- coding: utf-8 -*-
+"""
+Created on Mon Jan 14 22:01:43 2019
+@author: Florentin Popescu
+"""
+
+#===================LESSON_03====================
+# Mailroom, Part 1-------------------------------
+#================================================
+
+#================================================
+# Sending a Thank You note
+#================================================
+import re
+
+#donors on file
+donors = [['Name1', 1, 10, 100], 
+          ['Name2', 5.50, 15.50, 25.50],
+          ['Name3', 1000000],
+          ['Name5', 10.50, 20.75, 4.25],
+          ['Name6', 999.99, 99.99, 9.99],
+          ['Name7', 0.75,  0.25]]
+
+
+choices = ["Send a Thank You", "Create a Report", "Quit"]
+
+#-----------------------------------------------
+def donor_list(table):
+    return [item[0] for item in table]
+
+#-----------------------------------------------
+def add_donor(table, item):
+    return table.append([item])
+
+#-----------------------------------------------
+def check_name(string):
+    '''
+    Test if donor's name contains special characters
+    '''
+    regex = re.compile("[-+=@_!#$%^&*()<>?/\|}{~:]")
+    if(regex.search(string) == None): 
+        return True 
+    else: 
+        return False
+
+#-----------------------------------------------
+def statistics(table):
+    '''
+    Compute base statisting on donations (sum, average).
+    '''
+    #gardians - avoid division by 0
+    tbl = [[item[0], sum(item[1:]), len(item[1:]), 
+           sum(item[1:])/len(item[1:])] for item in table if item[1] > 0]
+    tbl.sort(key=lambda x: x[1], reverse = True)
+    return tbl
+
+#-----------------------------------------------
+def letter(name, amount):
+    '''
+    Print a costumized thank-you note.
+    '''
+    print(f"\nDear {name.lstrip().rstrip():s}, \nThank you for you donation of ${amount:,.2f} to our charity.")
+
+#-----------------------------------------------
+def add_donation(table, name, amount):
+    return table[donor_list(table).index(name)].append(amount)
+    
+#-----------------------------------------------
+def addDonation_sendLetter(table, x, amount):

```

Comment:
all lower case for function names, also don't use `x` for var names in methods especiallys